File: src/main/java/sage/transfer/HotTopic.java
package sage.transfer;

import java.util.Date;

import sage.entity.TopicPost;

public class HotTopic implements Comparable<HotTopic> {
  public long id;
  public TopicPreview topic;
  public int replyCount;
  public Date lastActiveTime;
  public double rank;

  HotTopic() {}

  public HotTopic(TopicPreview topic, int replyCount, Date lastReplyTime) {
    id = topic.getId();
    this.topic = topic;
    this.replyCount = replyCount;
    lastActiveTime = lastReplyTime!=null ? lastReplyTime : topic.getWhenCreated();
  }

  @Override
  public int compareTo(HotTopic o) {
    return -Double.compare(rank, o.rank);
  }
}

File: src/main/kotlin/sage/transfer/HotTopic.kt
package sage.transfer

import java.util.Date

class HotTopic : Comparable<HotTopic> {
  var id: Long = 0
  var topic: TopicPreview? = null
  var replyCount: Int = 0
  var lastActiveTime: Date? = null
  var rank: Double = 0.0

  internal constructor() {
  }

  constructor(topic: TopicPreview, replyCount: Int, lastReplyTime: Date?) {
    id = topic.id
    this.topic = topic
    this.replyCount = replyCount
    lastActiveTime = lastReplyTime ?: topic.whenCreated
  }

  override fun compareTo(other: HotTopic): Int {
    return -java.lang.Double.compare(rank, other.rank)
  }
}
